### PR TITLE
add config options to switch between http and https + support the kube-lego magic ingress label

### PR DESCRIPTION
**Pull request by rawlingsj**

### CODE REVIEW

**Comment by fabric8cd** [comment]
@rawlingsj snapshot exposecontroller image is available for testing.  `docker pull docker.io/fabric8/exposecontroller:SNAPSHOT-PR-61-5`

**Comment by fabric8cd** [comment]
@rawlingsj snapshot exposecontroller image is available for testing.  `docker pull docker.io/fabric8/exposecontroller:SNAPSHOT-PR-61-4`

File: exposestrategy/ingress.go [hunk 1]
```diff
@@ -71,6 +76,9 @@ func (s *IngressStrategy) Add(svc *api.Service) error {
 		ingress.Labels = map[string]string{}
 	}
 	ingress.Labels["provider"] = "fabric8"
+	if s.tlsAcme {
+		ingress.Labels["kubernetes.io/tls-acme"] = "true"
```
Comment (rasheedamir) [inline exposestrategy/ingress.go]: @rawlingsj shouldn't this be an annotation and not a label?

Comment (rawlingsj) [inline exposestrategy/ingress.go]: Good spot thanks!

**Comment by fabric8cd** [comment]
@rawlingsj snapshot exposecontroller image is available for testing.  `docker pull docker.io/fabric8/exposecontroller:SNAPSHOT-PR-61-5`

**Comment by fabric8cd** [comment]
@rawlingsj snapshot exposecontroller image is available for testing.  `docker pull docker.io/fabric8/exposecontroller:SNAPSHOT-PR-61-6`

**Comment by fabric8cd** [comment]
@rawlingsj snapshot exposecontroller image is available for testing.  `docker pull docker.io/fabric8/exposecontroller:SNAPSHOT-PR-61-7`

**Comment by fabric8cd** [comment]
@rawlingsj snapshot exposecontroller image is available for testing.  `docker pull docker.io/fabric8/exposecontroller:SNAPSHOT-PR-61-8`

**Comment by fabric8cd** [comment]
@rawlingsj snapshot exposecontroller image is available for testing.  `docker pull docker.io/fabric8/exposecontroller:SNAPSHOT-PR-61-9`

**Pull request merged**